connect/transforms/src/test/java/org/apache/kafka/connect/transforms/InsertHeaderTest.java

```diff
@@ -0,0 +1,233 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.connect.transforms;
+
+import org.apache.kafka.connect.data.SchemaAndValue;
+import org.apache.kafka.connect.data.Time;
+import org.apache.kafka.connect.data.Timestamp;
+import org.apache.kafka.connect.header.ConnectHeaders;
+import org.apache.kafka.connect.header.Headers;
+import org.apache.kafka.connect.source.SourceRecord;
+import org.junit.After;
+import org.junit.Test;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+
+public class InsertHeaderTest {
+    private InsertHeader<SourceRecord> xform = new InsertHeader<>();
+
+    @After
+    public void teardown() {
+        xform.close();
+    }
+
+    @Test
+    public void insertConfiguredHeaderWithStringValue() {
```

Comment:
Let's add separate tests that attempt to call the SMT on null record, on a record that has some headers but none that match the names, and a record that has some headers that do and do not match the STM headers, all asserting that we added the header, even if there was one with a matching name.

Reply:
Would also be good to add a test that asserted that the SMT configure fails if the name is not valid (e.g., blank or null).

Reply:
One more test to add: configure and use an SMT that adds a header with a null value to the record.